### ISSUE TITLE
feature/datamodel: when `import: FooBar` where `FooBar.yaml` doesn't exist, it crashes

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#112]
@PeterTKY can you fix it so that the code provides a warning or error describing the problem, instead of crashing? Thanks.

**Comment by PeterTKY** [comment]
Fixed by c2fc2583e1198fdc87ccbd17d537e3fbd8cf5f84

It will show warning messages, but document will continue to compile.

**Comment by ronaldtse** [comment]
Thank you Peter!

**Comment by ronaldtse** [comment]
Still crashing:

```sh
FILENAME=sources/T-REC-E.999.adoc; \
	echo "Compiling locally..."; bundle exec metanorma $FILENAME
Compiling locally...
asciidoctor: ERROR: sources/sections/08-timezone-shift-rule.adoc: line 31: include file not found: /Users/me/src/calconnect/csd-timezone-model/sources/models/views/applicability-rule-view.yml
bundler: failed to load command: metanorma (/Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/bin/metanorma)
NoMethodError: asciidoctor: FAILED: sources/T-REC-E.999.adoc: Failed to load AsciiDoc document - undefined method `reduce' for nil:NilClass
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/bundler/gems/metanorma-standoc-c2fc2583e119/lib/asciidoctor/standoc/datamodel_block_macro.rb:142:in `process_imported_models'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/bundler/gems/metanorma-standoc-c2fc2583e119/lib/asciidoctor/standoc/datamodel_block_macro.rb:29:in `process'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:1038:in `[]'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:1038:in `build_block'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:866:in `next_block'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:369:in `next_section'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:354:in `next_section'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:354:in `next_section'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:354:in `next_section'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/parser.rb:98:in `parse'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor/document.rb:570:in `parse'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor.rb:1367:in `load'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/asciidoctor-1.5.8/lib/asciidoctor.rb:1479:in `convert'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-0.3.14/lib/metanorma/input/asciidoc.rb:8:in `process'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-itu-0.2.4/lib/metanorma/itu/processor.rb:26:in `input_to_isodoc'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-0.3.14/lib/metanorma/compile.rb:101:in `process_input'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-0.3.14/lib/metanorma/compile.rb:17:in `compile'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/lib/metanorma/cli/compiler.rb:25:in `compile_file'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/lib/metanorma/cli/compiler.rb:13:in `compile'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/lib/metanorma/cli/compiler.rb:17:in `compile'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/lib/metanorma/cli/command.rb:34:in `compile'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/lib/metanorma/cli.rb:59:in `start'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/metanorma-cli-1.2.4/exe/metanorma:20:in `<top (required)>'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/bin/metanorma:23:in `load'
  /Users/me/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/bin/metanorma:23:in `<top (required)>'
```

**Comment by PeterTKY** [comment]
It looks to be another problem. In line 142:
`
view_hash["imports"].reduce(view_hash) do |acc, (model_path, model_fidelity)|
`

The reason why it crashes is `view_hash["imports"]` is nil. It's not related to whether or not the file is found. A possible reason I can think of is that `import` or other typo instead of `imports` was defined in the view yaml...

Will fix the nil issue by assuming an empty Hash instead of nil but it won't fix the bug of the typo.

**Comment by PeterTKY** [comment]
Fixed with 8fe38e34db9f984ebd74d69ce4e53f7a2d80171c.
